src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,179 @@
+
+
+if typing.TYPE_CHECKING:
+    from mypy_boto3_sts.client import STSClient
+    from mypy_boto3_sts.type_defs import AssumeRoleResponseTypeDef
+
+from .plugin import CredentialPlugin
+
+
+_aws_cred_cache: dict[
+    str,
+    CredentialsTypeDef | dict[typing.Never, typing.Never],
+] | dict[typing.Never, typing.Never] = {}
+
+
+assume_role_inputs = {
+    'fields': [
+        {
+            'id': 'access_key',
+            'label': _('AWS Access Key'),
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS access key for'
+                'the user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'secret_key',
+            'label': 'AWS Secret Key',
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS secret key for the'
+                'user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'external_id',
+            'label': 'External ID',
+            'type': 'string',
+            'help_text': _(
+                'The optional External ID which will'
+                'be provided to the assume role API.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'identifier',
+            'label': 'Identifier',
+            'type': 'string',
+            'help_text': _(
+                'The name of the key in the assumed AWS role'
+                'to fetch [AccessKeyId | SecretAccessKey | SessionToken].',
+            ),
+        },
+    ],
+    'required': [
+        'role_arn',
+    ],
+}
+
+
+def aws_assumerole_getcreds(
+        access_key: str | None,
+        secret_key: str | None,
+        role_arn: str | None,
+        external_id: int,
+) -> CredentialsTypeDef | dict[typing.Never, typing.Never]:
+    """Return the credentials for use.
+
+    :param access_key: The AWS access key ID.
+    :type access_key: str
+    :param secret_key: The AWS secret access key.
+    :type secret_key: str
+    :param role_arn: The ARN received from AWS.
+    :type role_arn: str
+    :param external_id: The external ID received from AWS.
+    :type external_id: int
+    :returns: The credentials received from AWS.
+    :rtype: dict
+    :raises ValueError: If the client response is bad.
+    """
+    connection: STSClient = boto3.client(
+        service_name='sts',
+        # The following EE creds are read from the env if they are not passed:
+        aws_access_key_id=access_key,  # defaults to `None` in the lib
+        aws_secret_access_key=secret_key,  # defaults to `None` in the lib
+    )
+    try:
+        response: AssumeRoleResponseTypeDef = connection.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='AAP_AWS_Role_Session1',
+            ExternalId=external_id,
+        )
+    except ClientError as client_err:
+        raise ValueError(
```

Comment:
`ValueError` does not seem like a good fit for this kind of error. It's probably a `RuntimeError` or something like that.
```suggestion
        raise RuntimeError(
```
Or perhaps a
```suggestion
        raise LookupError(
```
It might also be a
```suggestion
        raise ConnectionError(
```

`ValueError` would signal to the caller that they passed something wrong into the function call. But this message seems to imply it's an unexpected platform problem or something like that.

Reply:
I think any of these make sense and agree value isn't the correct thing in place. Only one I can see kinda not fitting is connection since errors could or could not be connection related. IMO lookup makes the most since we the actual function call is going and "lookin " something up but I am not dead set on that.

Reply:
Yep. That's what I often use. And the calling code would need to handle it and turn into whatever awx interface currently wants, which seems to be a `ValueError`, looking into the function below.